src/us/kbase/workspace/kbase/IdentifierUtils.java

```diff
@@ -267,12 +264,21 @@ private static ObjectIdentifier buildObjectIdentifier(
 					"Error on ObjectSpecification #"
 					+ objcount + ": " + e.getLocalizedMessage(), e);
 		}
-		if (paths == null && refchain == null) {
+		final boolean searchDAG = longToBoolean(o.getFindReferencePath());
+		if (paths == null && refchain == null && !searchDAG) {
 			return oi;
 		} else if (paths == null) {
-			return new ObjectIDWithRefPath(oi, refchain);
+			if (searchDAG) {
+				return new ObjectIDWithRefPath(oi);
+			} else {
+				return new ObjectIDWithRefPath(oi, refchain);
```

Comment:
No, it's checked in earlier places in the code in the mutateObjSpecByRefString method and in the processRefChains method. At this point it's guaranteed that if searchDAG is true no paths were provided.